distribution/lib/Standard/Base/0.0.0-dev/src/System/File/Posix_File_Permission.enso

```diff
@@ -0,0 +1,41 @@
+from Standard.Base import all
+
+polyglot java import java.nio.file.attribute.PosixFilePermission
+
+## Enso model of the Java `PosixFilePermission` enum.
+type Posix_File_Permission
```

Comment:
Would a type:
```
type Posix_File_Permission
    type Posix_File_Permission mod:Integer

    owner_read : Boolean
    owner_read = (self.mod / 100).bit_and 4

    owner_write : Boolean
    owner_write = (self.mod / 100).bit_and 2

    owner_execute : Boolean
    owner_execute = (self.mod / 100).bit_and 1
```

Seems a better way to represent than an array of flags?

Reply:
I don't think so.

Firstly, it seems quite weird to me to use our decimal numbers where in reality the modes are octal. Of course it can work, but sounds to me like a source of potential bugs (someone wanting to use the number where the octal would really be expected).

Also, it adds complexity which I'm not sure is needed - the Java methods return a Set of options represented as an enum and I wanted to reflect that here. We need a logic for converting back-and-forth between Enso and Java representations. With the current one its trivial and straightforward, with this proposal it's easy but more complex than what we have now.

I don't think we should expose close to the user any kind of bit-fiddling unless really needed - Enso is a higher-level language after all.

Reply:
If you think the current big enum is bad (I'm not a big fan either), I'd suggest maybe this architecture:

```
type Permission
    type Read
    type Write
    type Execute

type File_Permissions (owner : Vector Permission) (group : Vector Permission) (other : Vector Permission)
```

What do you think? Shall I move to it or something similar?

Reply:
Sorry I meant integer division: 

```python
type Posix_File_Permission
    type Posix_File_Permission mod:Integer

    owner_read : Boolean
    owner_read = (self.mod.div 100)%10 . bit_and 4

    owner_write : Boolean
    owner_write = (self.mod.div 100)%10 . bit_and 2

    owner_execute : Boolean
    owner_execute = (self.mod.div 100)%10 . bit_and 1

    group_read : Boolean
    group_read = (self.mod.div 10)%10 . bit_and 4

    group_write : Boolean
    group_write = (self.mod.div 10)%10 . bit_and 2

    group_execute : Boolean
    group_execute = (self.mod.div 10)%10 . bit_and 1

    other_read : Boolean
    other_read = self.mod%10 . bit_and 4

    other_write : Boolean
    other_write = self.mod%10 . bit_and 2

    other_execute : Boolean
    other_execute = self.mod%10 . bit_and 1
```

Agree we don't expect user to do bit fiddling but felt like reasonable to do it within our class.

Also prefer the 3 Vectors as seems definitely cleaner.

Reply:
I just don't see a point of this bit-based representation because when talking with the outside world we are using the enums anyway.

I'll do the 3 vectors then

Reply:
Just seemed easier to work with `File.new "Hello.txt" . posix_permissions . owner_read` felt cleaner than `.contains` on a Vector.

Reply:
Ah, I see.

Well I can add getters to the new `File_Permissions` like this.
So then I could do
```
owner_read : Boolean
owner_read = self.owner.contains Read
```

Does that sounds good?

Reply:
Pushed the updated implementation